proposals/NNNN-new-remote-control-modules-and-parameters.md

```diff
@@ -0,0 +1,512 @@
+
+- Some light control applications want to change the ambient lights of the vehicle and/or turn on/off certain exterior lights (while the vehicle is in parking).
+
+- Some radio control applications need more than the traditional RDS data, they need HD radio status data (especially the station id in the SIS data).
+
+
+## Proposed solution
+
+As usual, the remote control mobile applications use the following RPCs
+- `GetSystemCapability` with `SystemCapabilityType=REMOTE_CONTROL` to read available control modules and available control items within each module;
+- `GetInteriorVechileData` to read the current status value of a remote-control module;
+- `GetInteriorVechileData` with `subscribe`=true/false to register/un-register a module's data change notifications;
+- `SetInteriorVechileData` to change the settings/status of a remote-control module;
+- `OnInteriorVechileData` to receive any remote-control module value change notifications;
+
+In addition, in a GetInteriorVechileData response or OnInteriorVechileData notifications, an `xyzmoduleControlData` indicates the current value of the module status; in SetInteriorVechileData request, an `xyzmoduleControlData` indicates the required target state of the module.
+
+#### Add there new module types to the existing module list.
+| RC Module |
+| --------- |
+| Audio |
+| Light |
+| HMI Settings |
+
+#### For "AUDIO" remote-control mobile application will be able to READ and SET:
+
+| Control Item | Value Range |Type | Comments |
+| ------------ |------------ | ------------ | ------------ |
+| Audio volume | 0%-100%| Get/Set/Notification | The audio source volume level |
+| Audio Source | MOBILE_APP, RADIO_TUNER, CD, BLUETOOTH, USB, etc. see PrimaryAudioSource| Get/Set/Notification | defines one of the available audio sources |
+| keep Context | true, false| Set only | control whether HMI shall keep current application context or switch to default media UI/APP associated with the audio source|
+| Equilizer Settings | Struct {Channel ID as integer, Channel setting as 0%-100%} | Get/Set/Notification | Defines the list of supported channels (band) and their current/desired settings on HMI
+
+Any remote control applications can read the current audio data. A RC and media application can set audio source to either itself (with target `MOBILE_APP`) or other system sources. It CANNOT set audio source to other mobile applications. When a mobile application sends a SetInteriorVechileData request to change the audio source from MOBILE_APP to other types of audio source without `keepContext` parameter or with `keepContext=false`, after the system successfully executes the request, the application will go to HMI level 'BACKGROUND'. If in the same request `keepContext`=true, the system shall not change this application's HMI level.
```

Comment:
Please update to `SetInteriorVehicleData`

Reply:
changed